File: Week3/589.py
"""
https://leetcode-cn.com/problems/n-ary-tree-preorder-traversal/
非递归解法
"""
class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        if root:
            res=[]
            queue=[root]
            while queue:
                item=queue.pop()
                res.append(item.val)
                if item.children:
                    queue.extend(item.children[::-1])
            return res
        return

File: Week3/589.java
class Solution {
    private List<Integer> ans=new ArrayList<>();
    public List<Integer> preorder(Node root) {
        PreOrder(root);
        return ans;
    }
    private void PreOrder(Node root){
        if (root!=null){
            ans.add(root.val);
            for(Node child:root.children){
                PreOrder(child);
            }
        }
    }
}